types/rdf-ext/rdf-ext-tests.ts

```diff
@@ -246,6 +246,26 @@ function dataset_match() {
     matched = rdf.dataset().match(null, null, null, term);
 }
 
+function test_reduce() {
+    const quad: QuadExt = <any> {};
+    const dataset: DatasetExt = <any> {};
+
+    // $ExpectType QuadExt
+    const reduced = dataset.reduce((previousValue: QuadExt, currentValue: QuadExt, currentIndex: number, self: DatasetExt) => {
+        return previousValue;
+    });
+
+    // $ExpectType QuadExt
+    const reducedWithInitial = dataset.reduce((previousValue: QuadExt, currentValue: QuadExt, currentIndex: number, self: DatasetExt) => {
+        return previousValue;
+    }, quad);
+
+    // $ExpectType string
+    const reducedToOther = dataset.reduce((previousValue: string, currentValue: QuadExt, currentIndex: number, self: DatasetExt) => {
+        return previousValue;
+    }, '');
```

Comment:
Perhaps include a test with an explicit type parameter for reduce?

Reply:
Committed. Is that what you meant?

Reply:
Yup - approved :)